packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -417,6 +417,27 @@ class FileCollection extends DocumentCollection {
     return this.stackClient.fetchJSON('POST', uri`/files/trash/${id}`)
   }
 
+  /**
+   * Copy a file.
+   *
+   * @param {string} id   - The file's id
+   * @param {string} name   - The file copy name
+   * @param {string} dirId   - The destination directory id
+   * @returns {Promise}   - A promise that returns the copied file if resolved.
+   * @throws {FetchError}
+   *
+   */
+  copy(id, name, dirId) {
+    const params = {
+      name: name === undefined ? undefined : sanitizeAndValidateFileName(name),
+      dirId
+    }
+    const path = uri`/files/${id}/copy`
+    const url = querystring.buildURL(path, params)
+
+    return this.stackClient.fetchJSON('POST', url)
```

Comment:
At first, when reading your code, I though that the stack doesn't return any content. But this is not the case : https://docs.cozy.io/en/cozy-stack/files/#response_9 

If its a success, then the stack respond with the new created document. And this is important to deal with it in CozyClient even if for this method, this is not very important **yet**. 

I will not take the time to write a fully detailed comment but: 
1/ If we use the DSL and not make a request through the Collection directly, we await the response of the request
2/ The, we dispatched it to the cozy-client's redux store. 
3/ Redux store is updated with this new document (without making any stack request) 
4/ Cozy-Client replays all the queries, in memory aka without requesting the stack (through http) , already made and add this new document to the result 
5/ All your concerned queries by this document will be updated 
6/ You'll have your application up to date without making any http request 

But why does it works in Drive, because it works and there is no http request to get the result? Good question @zatteo! 

There is something called `realtime` on Cozy : https://docs.cozy.io/en/cozy-stack/realtime/ . When something happens in CouchDB, the stack send a message to say the change. 

Cozy-Drive is linked to this realtime by using the `RealtimeQueries` component: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/RealTimeQueries.jsx like what we do in "2/". 

This component listen for a message within the websocket, and when there is something, dispatch the result like what we do in "2/". 

So why do the works twice? 

As I said, the mechanism from the redux store directly only works when we do not bypass it. We bypass it by calling the collection directly. We need to work on making accessible those method through the DSL: https://github.com/cozy/cozy-client/pull/650 

But we also need to rely on the realtime for drive: There is a lot of application that can manipulate files on your cozy (konnectors, but also other apps). And in that case, cozy-client can't handle that for you since this is not the same instance. 

And I do not want to exclusively rely on the realtime because it adds latence (since stack sends a message). Message can be missed and so one.

Reply:
So we shoudl wait the response, normalize it and return the normalized data. 

Like that, you can also change the return type to `FileDocument`